packages/nimble-components/src/select/tests/select.stories.ts

```diff
@@ -73,6 +75,10 @@ The act of filtering will use the \`hidden\` attribute on the options to remove
 It is recommended that if the \`Select\` has 15 or fewer options that you use the \`none\` setting for the \`filter-mode\`.
 `;
 
+const placeholderDescription = `
+To display placeholder text within the \`Select\` you must provide an option before all other options that has both the \`disabled\` and \`selected\` attributes initially set. This option will not be available in the dropdown, and its contents will be used as the placeholder text.
```

Comment:
What benefit do we get from specifying a placeholder via an impossible combination of existing attributes (`disabled` and `selected`) vs adding a new attribute? To me it seems like a somewhat convoluted way to achieve this outcome.

When we were talking earlier I thought the benefit was that it matched the native select API. But from the spec link you added, the only part that I see being aligned is that the placeholder is specified by configuring the first option. So could we achieve the same thing more explicitly with something like this?

```html
<nimble-select>
  <nimble-list-option placeholder>Select an item</nimble-list-option>
  <nimble-list-option>First</nimble-list-option>
  <nimble-list-option>Second</nimble-list-option>
</nimble-select>
```

We could enforce (or document but not enforce) whatever rules we wanted like:
1. `placeholder` can only be added to one option
2. `placeholder` must be added to the first option
3. `placeholder` shouldn't be used with `disabled` or `selected`

Reply:
At the moment I have nothing against this approach.

Reply:
Since [`placeholder` is a built-in HTML attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/placeholder) we might decide to use something custom like `is-placeholder` to follow our [attribute naming convention](https://github.com/ni/nimble/blob/877f01c4512dccfe59767c82806467861460b227/packages/nimble-components/CONTRIBUTING.md#attribute-naming-convention).

Reply:
> What benefit do we get from specifying a placeholder via an impossible combination of existing attributes (disabled and selected) vs adding a new attribute?

Because it aligns with conventions in HTML and we should follow that over inventing new apis.

Reply:
Openui is making the new selectmenu element and this is their current idea: https://github.com/openui/open-ui/issues/637#issuecomment-1682780451

> RESOLVED: Support the value="" attribute, selected attribute, disabled attribute, and hidden attribute on option elements in selectlist. For v2, consider a placeholder attribute or element that implies <option value="" disabled hidden>.

Reply:
I didn't see anything about the `disabled` and `selected` parts of that API in the spec link in the PR description:
>If a select element has a required attribute specified, does not have a multiple attribute specified, and has a display size of 1; and if the value of the first option element in the select element's list of options (if any) is the empty string, and that option element's parent node is the select element (and not an optgroup element), then that option is the select element's placeholder label option.

Open UI is a good guide to follow. If we can include links to that and to the place the spec specifies `disabled` and `selected`, then I'm fine to follow that convention.

Reply:
> I didn't see anything about the disabled and selected parts of that API in [the spec link in the PR description](https://html.spec.whatwg.org/multipage/form-elements.html#attr-select-required)

Sorry, you're right spec didn't show it because that convention for placeholder described in the spec allows the placeholder to be selected.

Our usage guidance would be more like the openui resolution to configure:
- required + value="": causes it to fail HTML 5 form validation
- disabled: prevents it from being selectable but also prevents it from being the initial value
- selected: forces it to be selected even though it is disabled
- hidden: hides it in the option list

Reply:
The API that started out seeming convoluted makes more sense when broken down into those pieces. Thanks!

This brings up another question: right now the implementation is has special cases to detect this combination of attributes and behave differently. Should/could we instead implement each of the behaviors independently and have the placeholder behavior fall out naturally?

Reply:
I think your suggestion is aimed at responding to changes to the options themselves, and if one ends up in the state that is indicative of the placeholder, that it _becomes_ the placeholder (presuming there is no option before it that currently represents the placeholder). I have nothing against this. I think this is an additive change to the existing implementation.

Reply:
Very apropos 
![Screenshot_20240221-185226](https://github.com/ni/nimble/assets/1588923/1ad2216e-7b32-4c33-aeb9-c60977e18c62)

Reply:
I have implemented the API in such a way that the "placeholder" option more-or-less just falls out naturally from the API configuration. The `Select` still must know that its current displayed option is the "placeholder" option so that it can be styled accordingly, and thus there is some implementation in place to make that possible.